qa/standalone/crush/erasure-code-profile.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2020 Red Hat <contact@redhat.com>
+#
+# Author: Prashant Dhange <pdhange@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Library Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library Public License for more details.
+#
+
+source $CEPH_ROOT/qa/standalone/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:7132" # git grep '\<7132\>' : there must be only one
+    export CEPH_ARGS
+    CEPH_ARGS+="--fsid=$(uuidgen) --auth-supported=none "
+    CEPH_ARGS+="--mon-host=$CEPH_MON "
+
+    local funcs=${@:-$(set | sed -n -e 's/^\(TEST_[0-9a-z_]*\) .*/\1/p')}
+    for func in $funcs ; do
+        setup $dir || return 1
+        $func $dir || return 1
+        teardown $dir || return 1
+    done
+}
+
+function TEST_valid_failuredomain() {
+    local dir=$1
+
+    run_mon $dir a || return 1
+
+    local profile=validecprofile
+
+    crush_domain=datacenter
+    ceph osd erasure-code-profile set $profile k=4 m=2 crush-failure-domain=$crush_domain 2>&1 || return 1
+    ceph osd erasure-code-profile ls | grep $profile || return 1
+    ceph osd erasure-code-profile get $profile|grep "crush-failure-domain=$crush_domain" || return 1
+    ceph osd erasure-code-profile rm $profile
+
+    crush_domain=rack
+    ceph osd erasure-code-profile set $profile k=4 m=2 crush-failure-domain=$crush_domain 2>&1 || return 1
+    ceph osd erasure-code-profile ls | grep $profile || return 1
+    ceph osd erasure-code-profile get $profile|grep "crush-failure-domain=$crush_domain" || return 1
+    ceph osd erasure-code-profile rm $profile
+
+    crush_domain=host
+    ceph osd erasure-code-profile set $profile k=4 m=2 crush-failure-domain=$crush_domain 2>&1 || return 1
+    ceph osd erasure-code-profile ls | grep $profile || return 1
+    ceph osd erasure-code-profile get $profile|grep "crush-failure-domain=$crush_domain" || return 1
+    ceph osd erasure-code-profile rm $profile
+
+    crush_domain=osd
+    ceph osd erasure-code-profile set $profile k=4 m=2 crush-failure-domain=$crush_domain 2>&1 || return 1
+    ceph osd erasure-code-profile ls | grep $profile || return 1
+    ceph osd erasure-code-profile get $profile|grep "crush-failure-domain=$crush_domain" || return 1
+    ceph osd erasure-code-profile rm $profile
+}
+
+function TEST_invalid_failuredomain() {
+    local dir=$1
+
+    run_mon $dir a || return 1
+
+    local profile=invalidecprofile
+
+    crush_domain=foo
+    ! ceph osd erasure-code-profile set $profile k=4 m=2 crush-failure-domain=$crush_domain 2>&1 || return 1
```

Comment:
you are testing invalid "erasure-code-profile" or invalid "crush-failure-domain"?

Reply:
I am testing invalid crush-failure-domain here. Are you referring to function nomenclature as well as testcase filename ?

Reply:
but it seems you are also using an "invalidecprofile"? i interpret it as invalid ec profile. BTW, it deserves a better name, like "invalid_ec_profile".